### PR TITLE
test: Add GraphQL directive processing edge case tests

### PR DESCRIPTION
**Pull request by pekkah**
## Summary
- Comprehensive directive processing edge case tests covering @skip, @include, and custom directives
- Tests directive behavior with true/false conditions, variables, and complex expressions
- Tests directive combinations, precedence rules, and validation error scenarios
- Tests directive location validation, usage on fragments, nested fields, and enum values
- Tests @deprecated directive handling and custom directive recognition
- Tests argument validation, default values, and repeated directive detection
- Covers comprehensive directive processing scenarios and validation edge cases

## Test plan
- [x] Builds successfully with proper directive usage patterns
- [x] Comprehensive test coverage of directive processing scenarios
- [x] Uses correct GraphQL schema and resolver patterns
- [ ] All test cases pass (test expectations may need adjustment based on actual directive behavior)
- [ ] Tests provide comprehensive coverage of directive processing edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/GraphQL.Tests/DirectiveProcessingEdgeCasesFacts.cs [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+
+        var result = await Executor.Execute(schema, query);
+
+        Assert.NotNull(result);
+        Assert.NotNull(result.Errors);
+        Assert.Single(result.Errors);
+    }
+
+    [Fact]
+    public async Task Directive_OnSubscription_ShouldProcessCorrectly()
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/DirectiveProcessingEdgeCasesFacts.cs]: ✅ Fixed: Renamed misleading test from OnSubscription_ShouldProcessCorrectly to OnMultipleRootFields_ShouldProcessCorrectly since it tests standard query operations on multiple fields, not GraphQL subscriptions.

File: tests/GraphQL.Tests/DirectiveProcessingEdgeCasesFacts.cs [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+                { "fallback", context =>
+                {
+                    context.ResolvedValue = "fallback value";
+                    return ValueTask.CompletedTask;
+                }}
+            }
+        };
+
+        return await builder.Build(resolvers);
+    }
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/DirectiveProcessingEdgeCasesFacts.cs]: ✅ Fixed: Refactored schema creation methods to eliminate code duplication. CreateTestSchema now accepts an optional additionalSdl parameter, allowing reuse for different directive scenarios while maintaining the same base Query type and resolvers.

File: tests/GraphQL.Tests/DirectiveProcessingEdgeCasesFacts.cs [hunk 3]
```diff
@@ -0,0 +1,724 @@
...
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/DirectiveProcessingEdgeCasesFacts.cs]: ✅ Fixed: Removed CreateCustomDirectiveSchema method as suggested. Now using CreateTestSchema with the directive SDL parameter.
